wrapper/src/main/java/software/amazon/jdbc/plugin/limitless/LimitlessRouterServiceImpl.java

```diff
@@ -98,20 +141,30 @@ public List<HostSpec> forceGetLimitlessRouters(final String clusterId, final Pro
   }
 
   @Override
-  public void startMonitoring(final @NonNull PluginService pluginService,
-      final @NonNull HostSpec hostSpec,
+  public void startMonitoring(final @NonNull HostSpec hostSpec,
       final @NonNull Properties props,
       final int intervalMs) {
 
     try {
       final String limitlessRouterMonitorKey = pluginService.getHostListProvider().getClusterId();
       final long cacheExpirationNano = TimeUnit.MILLISECONDS.toNanos(MONITOR_DISPOSAL_TIME_MS.getLong(props));
+      final List<HostSpec> limitlessRouters = Collections.emptyList();
 
       limitlessRouterMonitors.computeIfAbsent(
           limitlessRouterMonitorKey,
           key -> this.limitlessRouterMonitorInitializer
-              .createLimitlessRouterMonitor(pluginService, hostSpec, props, intervalMs),
+              .createLimitlessRouterMonitor(
+                  pluginService,
+                  hostSpec,
+                  limitlessRouters,
+                  props,
+                  intervalMs),
           cacheExpirationNano);
+      limitlessRouterCache.computeIfAbsent(
```

Comment:
What's the purpose to add an empty list to the cache?

Reply:
Was trying something dumb. 
Updated so monitors have access to the router cache directly and can update it directly.